packages/synapse-react-client/stories/QueryWrapperPlotNav.stories.tsx

```diff
@@ -227,6 +231,44 @@ const handleCustomCommandClick = async (event: CustomControlCallbackData) => {
   // TODO: PORTALS-2682: event.refresh() should refresh the data but it currently doesn't
   event.refresh()
 }
+
+const handleAllDataCustomCommandClick = async (
+  event: CustomControlCallbackData,
+) => {
+  displayToast(
+    `Custom action applied to all results (see js console for more information)`,
+  )
+
+  const ids: string[] = []
+  if (event.request) {
+    console.log('Query request is available')
+    console.log(event.request)
+    // get all ids
+    event.request.query.sql = 'select id from syn51186974'
+    const results = await SynapseClient.getFullQueryTableResults(event.request)
+    results.queryResult?.queryResults.rows.map(row => {
+      if (row.values && row.values[0]) ids.push(row.values[0])
+    })
+  }
+
+  const localStorageFilter: ColumnSingleValueQueryFilter = {
+    concreteType:
+      'org.sagebionetworks.repo.model.table.ColumnSingleValueQueryFilter',
+    columnName: 'fileSynapseId',
+    operator: ColumnSingleValueFilterOperator.IN,
+    values: ids,
+  }
+  localStorage.setItem(
+    QUERY_FILTERS_LOCAL_STORAGE_KEY('syn51186974-selectedfiles'),
+    JSON.stringify([localStorageFilter]),
+  )
+  console.log(
+    'Local Storage value set, refresh table to see additionalFilter QueryFilter being utilized',
+  )
+  // TODO: PORTALS-2682: event.refresh() should refresh the data but it currently doesn't
```

Comment:
I see now, `refresh` just sends QueryWrapper the current query, so of course we get a cache hit and just see the same results. We really want this command to apply localStorage filters (likely by re-mounting the component).

Reply:
Added a way for the toplevelcontrols to remount the query wrapper (increments the key).  This does the trick!